tools/preprocessor.mjs

```diff
@@ -14,24 +14,28 @@
 
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
-global.vm = require('vm');
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vm from 'vm';
+import assert from 'assert';
+import * as url from 'url';
 
-const arguments_ = process.argv.slice(2);
+const args = process.argv.slice(2);
 const debug = false;
 
+global.vm = vm;
+global.assert = assert;
```

Comment:
Oh ..  maybe they are not here..  Removed

Reply:
Actually sorry both of these are needed by scripts that we import below.

`global.vm` is also set in the old code.   `global.assert` is added since in the non-module world assert exists by default on the global object.  In the module world you need to import it.